### DISCUSSION TITLE
Pre-route callback

### DISCUSSION THREAD

**Discussion by bombillazo** [nandorojo/solito#312] [Ideas]
This can be easily implemented with a custom component, but perhaps it could be useful in general:

How about a pre-route parameter/prop to run a callback before routing with push/\<Link />?

**Comment by nandorojo** [comment]
not sure i get what you mean

**Reply by bombillazo** [reply]
Like for example if we wanted to run a function before pushing to a route, having a callback parameter for `push`/`replace` or a prop for `<Link />`

```ts
const preFunc = () => { 
    // do something before push, like throw
  }

push(
  '/home',
  preFunc,
)

<Link preRoute={preFunc} />
```

Maybe its overkill 😅

**Reply by nandorojo** [reply]
I recommend using the `useLink` hook for those cases and extending the `onPress` function it returns
